src/csharp/Microsoft.Spark.E2ETest/IpcTests/ML/Feature/StopWordsRemoverTests.cs

```diff
@@ -0,0 +1,116 @@
+            string expectedInputCol = "input_col";
+            string expectedOutputCol = "output_col";
+            bool expectedCaseSensitive = false;
+            string[] expectedStopWords = new string[] {"test1", "test2"};
+            StructType expectedSchema = new StructType(new[]
+            {
+                new StructField("input_col", new ArrayType(new StringType(), true)),
+                new StructField("output_col", new ArrayType(new StringType(), true))
+            });
+
+            DataFrame input = _spark.Sql("SELECT split('Hi I heard about Spark', ' ') as input_col");
+
+            StopWordsRemover stopWordsRemover = new StopWordsRemover(expectedUid)
+                .SetInputCol(expectedInputCol)
+                .SetOutputCol(expectedOutputCol)
+                .SetCaseSensitive(expectedCaseSensitive)
+                .SetStopWords(expectedStopWords);
+
+            StructType outPutSchema = stopWordsRemover.TransformSchema(input.Schema());
+
+            DataFrame output = stopWordsRemover.Transform(input);
+            Assert.Contains(output.Schema().Fields, (f => f.Name == expectedOutputCol));
+            Assert.Equal(expectedInputCol, stopWordsRemover.GetInputCol());
+            Assert.Equal(expectedOutputCol, stopWordsRemover.GetOutputCol());
+            Assert.Equal(expectedCaseSensitive, stopWordsRemover.GetCaseSensitive());
+            Assert.Equal(expectedStopWords, stopWordsRemover.GetStopWords());
+            Assert.Equal(expectedSchema, outPutSchema);
+
+            using (TemporaryDirectory tempDirectory = new TemporaryDirectory())
+            {
+                string savePath = Path.Join(tempDirectory.Path, "StopWordsRemover");
+                stopWordsRemover.Save(savePath);
+
+                StopWordsRemover loadedStopWordsRemover = StopWordsRemover.Load(savePath);
+                Assert.Equal(stopWordsRemover.Uid(), loadedStopWordsRemover.Uid());
+            }
+
+            Assert.Equal(expectedUid, stopWordsRemover.Uid());
+
+            TestFeatureBase(stopWordsRemover, "inputCol", "input_col");
+        }
+
+        /// <summary>
+        /// Test stop words removers with locale, run if spark version is greater than spark 2.4.0
+        /// skip this test for rest of the spark versions.
+        /// </summary>
+        [SkipIfSparkVersionIsLessThan(Versions.V2_4_0)]
+        public void TestStopWordsRemoverWithLocale()
+        {
+            string expectedUid = "theUidWithLocale";
+            string expectedInputCol = "input_col";
+            string expectedOutputCol = "output_col";
+            string expectedLocale = "en_GB";
+            bool expectedCaseSensitive = false;
+            string[] expectedStopWords = new string[] {"test1", "test2"};
+
+            DataFrame input = _spark.Sql("SELECT split('Hi I heard about Spark', ' ') as input_col");
+
+            StopWordsRemover stopWordsRemover = new StopWordsRemover(expectedUid)
```

Comment:
In this test we don't need to re-run everything in the whole test, just create a new StopWordsRemover and the .SetLocale and .GetLocale